Core/include/Acts/Geometry/NavigationPolicyFactory.hpp

```diff
@@ -0,0 +1,220 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Navigation/INavigationPolicy.hpp"
+#include "Acts/Navigation/MultiNavigationPolicy.hpp"
+
+#include <concepts>
+#include <memory>
+namespace Acts {
+
+class TrackingVolume;
+class GeometryContext;
+class Logger;
+class INavigationPolicy;
+
+namespace detail {
+template <typename... Factories>
+class NavigationPolicyFactoryImpl;
+}
+
+/// Base class for navigation policy factories. The factory can be assembled
+/// iteratively by using `make` followed by a number of calls to the `add`
+/// function of the helper type. Example:
+///
+/// ```cpp
+/// auto factory = NavigationPolicyFactory::make()
+///  .add<NavigationPolicy1>(arg1, arg2)
+///  .add<NavigationPolicy2>(/*no args*/)
+///  .asUniquePtr();
+/// ```
+class NavigationPolicyFactory {
+ public:
+  virtual ~NavigationPolicyFactory() = default;
+
+  // This needs to be listed here, but the return type cannot be spelled out
+  // yet.
+  static auto make();
```

Comment:
alternatively this could be a free function but I guess there is no real advantage

Reply:
Right, seems equivalent.